src/Opdex.Platform.Infrastructure.Abstractions/Clients/CirrusFullNodeApi/Models/RawTransactionDto.cs

```diff
@@ -10,14 +13,61 @@ public RawTransactionDto()
         Vout = Array.Empty<VOutDto>();
     }
 
+    public Sha256 Hash { get; set; }
+
+    public VInDto[] Vin { get; set; }
+
     public VOutDto[] Vout { get; set; }
+
+    /*
+     *
+     * External transfers scriptPubKey include OP_CREATECONTRACT or OP_CALLCONTRACT (nonstandard)
+     * Internal transfers scriptSig include OP_SPEND (nonstandard)
+     * Internal transfers scriptPubKey include OP_INTERNALCONTRACTTRANSFER (nonstandard)
+     * Refund TXs are P2PKH
+     *
+     * Coinbase and SegWit transactions do not contain a script sig in the input. Therefore, we filter these out.
+     * We filter out transactions where the script sig uses nonstandard smart contract OP codes (internal transfers).
+     * Finally, we return transactions where the script pub key uses nonstandard smart contract OP codes.
+     *
+     */
+    public bool IsSmartContractCall => Vout.Any(v => v.ScriptPubKey.SmartContractScriptType == ScOpcodeType.OP_CALLCONTRACT);
+    public bool IsSmartContractCreate => Vout.Any(v => v.ScriptPubKey.SmartContractScriptType == ScOpcodeType.OP_CREATECONTRACT);
+
+    public bool IsExternalSmartContractTransfer => Vin.All(v => v.ScriptSig is not null && !v.ScriptSig.IsSmartContractScript);
```

Comment:
I know we've discussed this earlier but a small comment about this pointing to what you've shared with me could go a long way in the future if/when we need to come back to this code. 

Might be overkill but maybe not a bad idea to reference the previous links you've shared too:
https://github.com/stratisproject/StratisFullNode/blob/master/src/Stratis.SmartContracts.CLR/ContractExecutor.cs#L90
https://chainz.cryptoid.info/cirrus-test/block.dws?3434decc0e31d8e0a5cccb662b73ac225beaf259c7e8eceec69f83c71c69f7e2.htm

Reply:
Added some additional comments as you suggested